src/common/catalog/catalog-entity.ts

```diff
@@ -72,6 +83,32 @@ export abstract class CatalogCategory extends (EventEmitter as new () => TypedEm
   public getId(): string {
     return `${this.spec.group}/${this.spec.names.kind}`;
   }
+
+  /**
+   * Add a filter for menu items of catalogAddMenu
+   * @param fn The function that should return a truthy value if that menu item should be displayed
+   * @returns A function to remove that filter
+   */
+  public addMenuFilter(fn: AddMenuFilter): Disposer {
+    this.filters.add(fn);
+
+    return once(() => void this.filters.delete(fn));
```

Comment:
Only nit would be that if the category is removed, these should be removed too.

Reply:
Why? Do you mean if I call the `Disposer` returned by `CatalogCategoryRegistry::add`, I should remove the filters from the `CatalogCategory` instance? Doesn't sound right to me

Reply:
Yeah probably not.